lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -356,37 +356,51 @@ export class UserAgentApplication {
       scopes = this.filterScopes(scopes);
     }
 
-    this._loginInProgress = true;
-    
-    this.authorityInstance.ResolveEndpointsAsync()
-      .then(() => {
-        const authenticationRequest = new AuthenticationRequestParameters(this.authorityInstance, this.clientId, scopes, ResponseTypes.id_token, this._redirectUri, this._state);
-        if (extraQueryParameters) {
-          authenticationRequest.extraQueryParameters = extraQueryParameters;
-        }
-
-        var loginStartPage = this._cacheStorage.getItem(Constants.angularLoginRequest);
-        if (!loginStartPage || loginStartPage === "") {
-            loginStartPage = window.location.href;
-        }
-        else {
-            this._cacheStorage.setItem(Constants.angularLoginRequest, "")
-        }
-
-        this._cacheStorage.setItem(Constants.loginRequest, loginStartPage, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.loginError, "");
-        this._cacheStorage.setItem(Constants.stateLogin, authenticationRequest.state, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.nonceIdToken, authenticationRequest.nonce, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.msalError, "");
-        this._cacheStorage.setItem(Constants.msalErrorDescription, "");
-        const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
-        if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
-          this._cacheStorage.setItem(authorityKey, this.authority);
-        }
-
-        const urlNavigate = authenticationRequest.createNavigateUrl(scopes)  + Constants.prompt_select_account + Constants.response_mode_fragment;
-        this.promptUser(urlNavigate);
-      });
+      this._loginInProgress = true;
+      const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
+      var idTokenObject;
+      if(adalIdToken) {
+           idTokenObject = Utils.extractIdToken(adalIdToken);
+      }
+      if (idTokenObject && idTokenObject.upn ) {
```

Comment:
Check for consent scopes

Reply:
What if the user wants to pass the consent scopes in the login always as a calling pattern?  In that case unified cache will never get triggered. Unified cache is just one time flow when they migrate from adal to msal but after that when user gets the msal idtoken, unified cache flow may never get called again.

Reply:
WE agreed to ignore unified cache experience if user is passing consent scopes.